src/gui/widgets/settingswidget.cpp

```diff
@@ -357,3 +361,47 @@ void SettingsWidget::on_styleComboBox_currentTextChanged(const QString& new_styl
         }
     }
 }
+
+void SettingsWidget::on_fontComboBox_currentFontChanged(const QFont &f)
+{
+    qApp->setFont(f);
+    ASettings::write(ASettings::Main::Font, f.toString());
+}
+
+void SettingsWidget::on_fontSpinBox_valueChanged(int arg1)
+{
+    QFont f = qApp->font();
+    f.setPointSize(arg1);
+    qApp->setFont(f);
+    ASettings::write(ASettings::Main::FontSize, arg1);
+}
+
+void SettingsWidget::on_fontCheckBox_stateChanged(int arg1)
+{
+    switch (arg1) {
+    case Qt::Unchecked:
+    {
+        ui->fontComboBox->setEnabled(true);
+        ui->fontSpinBox->setEnabled(true);
+        ASettings::write(ASettings::Main::UseSystemFont, false);
+        QFont font(ui->fontComboBox->currentFont());
+        font.setPointSize(ui->fontSpinBox->value());
+        qApp->setFont(font);
+        break;
+    }
+    case Qt::Checked:
+    {
+        ui->fontComboBox->setEnabled(false);
+        ui->fontSpinBox->setEnabled(false);
+        ASettings::write(ASettings::Main::UseSystemFont, true);
+        QMessageBox message_box(this);
+        message_box.setText(tr("The application will be restarted for this change to take effect."));
+        message_box.exec();
+        qApp->quit();
+        QProcess::startDetached(qApp->arguments()[0], qApp->arguments());
```

Comment:
Im confused as to who knows where to find what font. When you set use system font you read the combobox on the right to write to the ini so its not system font its what the user chooses. You cant find the default with `QFont::defaultFont()` but i dont think its what the systems user has selected. It might be a fallback font for me its Deja Vu.

Reply:
Yeah that's a bit unclear.

When setting or unsetting using system font nothing is written to the combobox, so that the users last choice is saved and displayed to him, so you can easily switch between using the system font and 'your' custom font. You custom font is inactive (disabled / greyed out) but still available to be activated.

If the user never selected a custom font, then a fallback font is written to the ini, in your case Deja-Vu. Once you do select a custom font though, it should be read correctly.

The problems I had with the default font is that it is read at application startup, but if you set a custom font with `QApplication::setFont`, then this becomes the default font and the 'old' default font gets overwritten. That's also the reason that in order to revert to the system font, a restart is required.

I'll see if I can make the logic there a bit more intuitive.

I'll also add a warning for using a stylesheet in combination with a custom font, it's possible but there are a few catches.